tests/content-types.js

```diff
@@ -43,19 +30,177 @@ test('merged', t => {
+    t.fail('Merged should fail');
+  }).catch(e => {
+    t.is(e.message, 'Input \'text\' in content type \'Foo\' needs a name', 'Missing Name');
+  });
+});
+
+test('reject when name not found - id', t => {
+  const testCT = [{
+    name: 'Foo',
+    id: 'foo',
+    attributes: [
+      {
+        type: 'text',
+        id: 'My Text',
+      },
+    ],
+  }];
+
+  return types(testCT).then(result => {
+    t.fail('Merged should fail');
+  }).catch(e => {
+    t.is(e.message, 'Input \'My Text\' in content type \'Foo\' needs a name', 'Missing Name');
+  });
+});
+
+test('reject when id not found', t => {
+  const testCT = [{
+    name: 'Foo',
+    id: 'foo',
+    attributes: [
+      {
+        type: 'text',
+        name: 'My Text',
+      },
+    ],
+  }];
+
+  return types(testCT).then(result => {
+    t.fail('Merged should fail');
+  }).catch(e => {
+    t.is(e.message, 'Input \'My Text\' in content type \'Foo\' needs an ID', 'Missing ID');
+  });
+});
+
+test('reject when id is not kebab case', t => {
+  const testCT = [{
+    name: 'Foo',
+    id: 'foo',
+    attributes: [
+      {
+        type: 'text',
+        name: 'My Text',
+        id: 'myText',
+      },
+    ],
+  }];
+
+  return types(testCT).then(result => {
+    t.fail('Merged should fail');
+  }).catch(e => {
+    t.is(e.message, 'Input ID \'myText\' needs to be written in kebab case (e.g. \'my-text\')', 'Not Kebab ID');
+  });
+});
+
+test('reject when id is duplicated', t => {
+  const testCT = [{
+    name: 'Foo',
+    id: 'foo',
+    attributes: [
+      {
+        type: 'text',
+        name: 'My Text',
+        id: 'my-text',
+      },
+      {
+        type: 'email',
+        name: 'My Email',
+        id: 'my-text',
+      },
+    ],
+  }];
+
+  return types(testCT).then(result => {
+    t.fail('Merged should fail');
+  }).catch(e => {
+    t.is(e.message, 'Input ID \'my-text\' in content type \'Foo\' cannot be duplicated (in \'My Email\')', 'No Duplicate IDs');
+  });
+});
+
 
 test('merged with correct param', t => {
-  fooContentObj.id = 'foo-rific';
-  return types([fooContentObj])
+  const testCT = {
+    name: 'FooRific',
+    description: 'A very foo content model.',
+    id: 'foo-rific',
+    attributes: [
+      {
+        type: 'text',
+        id: 'username',
+        name: 'Username',
+        description: 'Please enter a username',
+      },
+      {
+        type: 'text',
+        id: 'full-name',
+        name: 'Full Name',
+        description: 'Please enter your full name',
+        inputs: {
+          text: {
+            label: 'Foo Bar Baz',
+          },
+        },
+      },
+    ],
+  };
+
+  return types([testCT])
     .then(result => {
+      // console.log(util.inspect(result, false, null));
       t.is(result[0].name, 'FooRific', 'Get first content type name');
       t.is(result[0].description, 'A very foo content model.', 'Get first content type desc');
       t.is(result[0].id, 'foo-rific', 'Get first content type id');
```

Comment:
if you want to overwrite label (see [comment above](https://github.com/punchcard-cms/content-types/pull/9/files#r64439926), you should add a `t.is` to check label too